frontend/src/pages/Confirmation/ConfirmationContainer.js

```diff
@@ -34,6 +34,9 @@ import { applyOriginalActions, createAdditionalActions } from "./createAdditiona
 import { executeOriginalActions } from "./executeOriginalActions";
 
 class ConfirmationContainer extends React.Component {
+  componentDidMount() {
+    this.props.cancelConfirmation(undefined);
```

Comment:
Why this is needed?

Reply:
This is needed to reset the redux state of Confirmation Dialog component. Bugs occurred after pressing the submit button and the API rejects the request - in this case, the submit-state remains true in redux store, hence the dialog is not showing up anymore (and the request are send immediately - without confirmation)

Solution: `cancelConfirmation()` only sets defaultState. For better readability, I should create another function called `resetConfirmation()` that also set the redux state to defaultState.
Another solution: Add action `CONFIRMATION_UNSUCESSFULL `and yield it on every API error related to confirmation dialog. This action resets redux state of confirmation dialog like this `state.set("confirmed", defaultState.get("confirmed"));`